### PR TITLE
Datachecks on Variation and VariationFeature

### PR DESCRIPTION
**Pull request by helensch**
Conversion and update of HealthCheck VariationFeature.java - see ENSVAR-1624
Conversion and update of HealthCheck VariationFeatureAlleles.java - see ENSVAR-1625

### CODE REVIEW

File: lib/Bio/EnsEMBL/DataCheck/Checks/VariationFeature.pm [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        AND vf2.seq_region_end = vf1.seq_region_end)
+  /;
+  is_rows_zero($self->dba, $sql_1, $desc_1);
+
+  # The MAF is also checked in Variation table
+  # TODO combine the MAF checks
+  my $desc_2 = 'VariationFeatures minor_allele_freq <= 0.5';
+  my $sql_2  = qq/
+    SELECT COUNT(*) 
+    FROM variation
```
[the remark this answers is not in the document]
Comment (helensch) [reply lib/Bio/EnsEMBL/DataCheck/Checks/VariationFeature.pm]: Thanks I have correct this

File: lib/Bio/EnsEMBL/DataCheck/Checks/VariationFeaturePAR.pm [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  GROUPS         => ['variation_tables'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['variation'],
+  TABLES         => ['variation_feature']
+};
+
+sub skip_tests {
+  my ($self) = @_;
+
+  return 1 if $self->species ne 'homo_sapiens';
```
[the remark this answers is not in the document]
Comment (helensch) [reply lib/Bio/EnsEMBL/DataCheck/Checks/VariationFeaturePAR.pm]: I have corrected the return value

File: lib/Bio/EnsEMBL/DataCheck/Checks/Variation.pm [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME        => 'Variation',
+  DESCRIPTION => 'Variation table data is present and correct',
+  GROUPS      => ['variation_tables'],
```
[the remark this answers is not in the document]
Comment (helensch) [reply lib/Bio/EnsEMBL/DataCheck/Checks/Variation.pm]: We need to review of our datacheck group names and which should be in the 'variation' group. We intend to add checks to pipelines so a group name for use in particular pipeline would be useful. So checks may need multiple groups.

[the remark this answers is not in the document]
Comment (james-monkeyshines) [reply lib/Bio/EnsEMBL/DataCheck/Checks/Variation.pm]: OK, we can discuss in more detail.

**Pull request merged**